src/org/parosproxy/paros/extension/manualrequest/ManualRequestEditorDialog.java

```diff
@@ -100,6 +101,10 @@ public ManualRequestEditorDialog(boolean isSendEnabled, String configurationKey)
 
 	protected void initialize() {
 		addWindowListener(new WindowAdapter() {
+			@Override
+			public void windowActivated(WindowEvent e) {
+				getMessageSender().cleanup();

```

Comment:
It would be better if the state of the button was tied to the state of the option (and this could be handled as a HTTP implementation detail). If you want to keep this I'd suggest adding a new (default) method to the interface with a name more accordingly to the behaviour provided (does not seem correct to "cleanup" each time the dialogue is focused, even though the WebSocket implementation is not doing anything).

Reply:
So you mean check the global option here and set the ‘selected’ value of the button (instead of calling cleanup)?

Reply:
It would not be here, in `HttpPanelSender`(or ManualHttpRequestEditorDialog?) by having it listen to the option change (`OptionsChangedListener`?) and update the button when that happens.